app/budget/services/actions/budgetActions.js

```diff
@@ -554,7 +572,34 @@ budgetApp.service('budgetActions', function ($rootScope, $window, budgetService,
 				$rootScope.$emit('toast', { message: "Could not save comment.", type: "ERROR" });
 			});
 		},
-		createLineItemComment: function (comment, lineItem, currentUserLoginId) {
+		// Will create the lineItem if necessary
+		createLineItemComment: function(comment, lineItem, currentUserLoginId) {
+			var self = this;
+
+			// Create lineItem if necessary
+			if (lineItem.id == null || lineItem.id <= 0) {
```

Comment:
Why doesn't line item exist before a line item comment is being created?

Reply:
The action is architected so that the view code can be responsible for conveying user behavior without having to also understand a lot of business logic (which would scatter business logic across directives, and involve repeating that logic). The action for creating a comment then handles ensuring the lineItem exists, and creating it if it does not.